ui/src/app/views/cohort-list/component.spec.ts

```diff
@@ -0,0 +1,107 @@
+import {FormsModule} from '@angular/forms';
+import {ActivatedRoute, UrlSegment} from '@angular/router';
+import {RouterTestingModule} from '@angular/router/testing';
+
+import {ClarityModule} from '@clr/angular';
+
+import {ProfileServiceStub} from 'testing/stubs/profile-service-stub';
+
+import {
+  updateAndTick
+} from '../../../testing/test-helpers';
+
+import {CohortsServiceStub} from 'testing/stubs/cohort-service-stub';
+import {WorkspacesServiceStub, WorkspaceStubVariables} from 'testing/stubs/workspace-service-stub';
+
+
+import {CohortListComponent} from '../cohort-list/component';
+
+import {
+  Cohort,
+  CohortsService,
+  WorkspaceAccessLevel
+} from 'generated';
+import {deepCopy} from '../../utils/index';
+
+const activatedRouteStub  = {
+  snapshot: {
+    url: [
+      {path: 'workspaces'},
+      {path: WorkspaceStubVariables.DEFAULT_WORKSPACE_NS},
+      {path: WorkspaceStubVariables.DEFAULT_WORKSPACE_ID}
+    ],
+    params: {
+      'ns': WorkspaceStubVariables.DEFAULT_WORKSPACE_NS,
+      'wsid': WorkspaceStubVariables.DEFAULT_WORKSPACE_ID
+    },
+    data: {
+      workspace: {
+        ...WorkspacesServiceStub.stubWorkspace(),
+        accessLevel: WorkspaceAccessLevel.OWNER,
+      }
+    }
+  }
+};
+
+class CohortListPage {
+  fixture: ComponentFixture<CohortListComponent>;
+  route: UrlSegment[];
+  form: DebugElement;
+
+  constructor(testBed: typeof TestBed) {
+    this.fixture = testBed.createComponent(CohortListComponent);
+    this.readPageData();
+  }
+
+  readPageData() {
+    updateAndTick(this.fixture);
+    updateAndTick(this.fixture);
+  }
+}
+
+describe('CohortListComponent', () => {
+  let cohortListPage: CohortListPage;
+  let profileServiceStub: ProfileServiceStub;
+
+  beforeEach(fakeAsync(() => {
+    profileServiceStub = new ProfileServiceStub();
+    TestBed.configureTestingModule({
+      imports: [
+        RouterTestingModule,
+        FormsModule,
+        ClarityModule.forRoot()
+      ],
+      declarations: [
+        CohortListComponent,
+      ],
+      providers: [
+        { provide: ActivatedRoute, useValue: activatedRouteStub},
+        { provide: CohortsService, useValue: new CohortsServiceStub()},
+      ] }).compileComponents().then(() => {
+      cohortListPage = new CohortListPage(TestBed);
+    });
+    tick();
+  }));
+
+  it('should create the app', fakeAsync(() => {
+    const fixture = TestBed.createComponent(CohortListComponent);
+    const app = fixture.debugElement.componentInstance;
+    updateAndTick(fixture);
+    expect(app).toBeTruthy();
+    expect(app.cohortList.length).toBe(2);
+  }));
+
+  it('should delete the correct cohort', fakeAsync(() => {
+    const fixture = TestBed.createComponent(CohortListComponent);
+    const app = fixture.debugElement.componentInstance;
+    updateAndTick(fixture);
+    const deletedCohort: Cohort = app.cohortList[0];
+    app.deleteCohort(deletedCohort);
```

Comment:
opt: My general preference in these tests is to click the UI and verify what is rendered after, rather than calling into internal methods and/or checking internal state.